privaterelay/management/commands/update_phone_remaining_stats.py

```diff
@@ -3,58 +3,88 @@
 from django.conf import settings
 from django.core.management.base import BaseCommand
 
-from allauth.socialaccount.models import SocialAccount
+import logging
+
+from emails.models import Profile
+from privaterelay.management.utils import (
+    get_free_phone_social_accounts,
+    get_phone_subscriber_social_accounts,
+)
 
 if settings.PHONES_ENABLED:
     from phones.models import RelayNumber
 
-from privaterelay.views import update_fxa
+logger = logging.getLogger("events")
+
+
+def reset_phone_remaining_stats(user):
+    # re-set remaining_texts and remaining_seconds to the maximum value
+    try:
+        relay_number = RelayNumber.objects.get(user=user)
+    except RelayNumber.DoesNotExist:
+        # no RelayNumber set, do nothing
+        return
+    relay_number.remaining_texts = settings.MAX_TEXTS_PER_BILLING_CYCLE
+    relay_number.remaining_seconds = settings.MAX_MINUTES_PER_BILLING_CYCLE * 60
+    relay_number.save()
 
 
-def update_phone_remaining_stats():
-    if not settings.PHONES_ENABLED:
-        return []
-    accounts_with_phones = []
-    for sub_with_phone in settings.SUBSCRIPTIONS_WITH_PHONE:
-        social_accounts = SocialAccount.objects.filter(
-            extra_data__icontains=sub_with_phone
+def get_next_reset_date(profile: Profile) -> datetime:
+    # TODO: consider moving this as a property in Profile model
+    # assumes that profile being passed have already been checked to have
+    # phone subscription or a free phone user
+    if profile.date_phone_subscription_reset is None:
+        # there is a problem with the sync_phone_related_dates_on_profile
+        # or a new foxfooder whose date_phone_subscription_reset did not get set in
+        logger.error(
+            "phone_user_profile_dates_not_set",
+            extra={
+                "fxa_uid": profile.fxa.uid,
+                "date_subscribed_phone": profile.date_phone_subscription_end,
+                "date_phone_subscription_start": profile.date_phone_subscription_start,
+                "date_phone_subscription_reset": profile.date_phone_subscription_reset,
+                "date_phone_subscription_end": profile.date_phone_subscription_end,
+            },
         )
-        accounts_with_phones.extend(list(social_accounts))
+        return datetime.now(timezone.utc) - timedelta(minutes=15)
+
+    calculated_next_reset_date = profile.date_phone_subscription_reset + timedelta(
+        settings.MAX_DAYS_IN_MONTH
+    )
+    if profile.date_phone_subscription_end is None:
+        return calculated_next_reset_date
+    if profile.date_phone_subscription_end < calculated_next_reset_date:
+        # return the past or the closest next reset date
+        return profile.date_phone_subscription_end
+    return calculated_next_reset_date
+
+
+def update_phone_remaining_stats() -> tuple[int, int]:
+    accounts_with_phones = get_phone_subscriber_social_accounts()
```

Comment:
suggestion (non-blocking): the other command calls this var `social_accounts`. could we change both to the most explicit var name `social_accounts_with_phones` ?